tests/lora/test_lora_layers_cogvideox.py

```diff
@@ -0,0 +1,150 @@
+
+from utils import PeftLoraLoaderMixinTests  # noqa: E402
+
+
+@require_peft_backend
+class CogVideoXLoRATests(unittest.TestCase, PeftLoraLoaderMixinTests):
+    pipeline_class = CogVideoXPipeline
+    scheduler_cls = CogVideoXDPMScheduler
+    scheduler_kwargs = {"timestep_spacing": "trailing"}
+
+    transformer_kwargs = {
+        "num_attention_heads": 4,
+        "attention_head_dim": 8,
+        "in_channels": 4,
+        "out_channels": 4,
+        "time_embed_dim": 2,
+        "text_embed_dim": 32,
+        "num_layers": 1,
+        "sample_width": 16,
+        "sample_height": 16,
+        "sample_frames": 9,
+        "patch_size": 2,
+        "temporal_compression_ratio": 4,
+        "max_text_seq_length": 16,
+    }
+    transformer_cls = CogVideoXTransformer3DModel
+    vae_kwargs = {
+        "in_channels": 3,
+        "out_channels": 3,
+        "down_block_types": (
+            "CogVideoXDownBlock3D",
+            "CogVideoXDownBlock3D",
+            "CogVideoXDownBlock3D",
+            "CogVideoXDownBlock3D",
+        ),
+        "up_block_types": (
+            "CogVideoXUpBlock3D",
+            "CogVideoXUpBlock3D",
+            "CogVideoXUpBlock3D",
+            "CogVideoXUpBlock3D",
+        ),
+        "block_out_channels": (8, 8, 8, 8),
+        "latent_channels": 4,
+        "layers_per_block": 1,
+        "norm_num_groups": 2,
+        "temporal_compression_ratio": 4,
+    }
+    vae_cls = AutoencoderKLCogVideoX
+    tokenizer_cls, tokenizer_id = AutoTokenizer, "hf-internal-testing/tiny-random-t5"
+    text_encoder_cls, text_encoder_id = T5EncoderModel, "hf-internal-testing/tiny-random-t5"
+
+    text_encoder_target_modules = ["q", "k", "v", "o"]
+
+    output_identifier_attribute = "frames"
+
+    @property
+    def output_shape(self):
+        return (1, 9, 16, 16, 3)
+
+    def get_dummy_inputs(self, with_generator=True):
+        batch_size = 1
+        sequence_length = 16
+        num_channels = 4
+        num_frames = 9
+        num_latent_frames = 3  # (9 - 1) // temporal_compression_ratio + 1
+        sizes = (2, 2)
+
+        generator = torch.manual_seed(0)
+        noise = floats_tensor((batch_size, num_latent_frames, num_channels) + sizes)
+        input_ids = torch.randint(1, sequence_length, size=(batch_size, sequence_length), generator=generator)
+
+        pipeline_inputs = {
+            "prompt": "dance monkey",
+            "num_frames": num_frames,
+            "num_inference_steps": 4,
+            "guidance_scale": 6.0,
+            # Cannot reduce because convolution kernel becomes bigger than sample
+            "height": 16,
+            "width": 16,
+            "max_sequence_length": sequence_length,
+            "output_type": "np",
+        }
+        if with_generator:
+            pipeline_inputs.update({"generator": generator})
+
+        return noise, input_ids, pipeline_inputs
+
+    def test_lora_fuse_nan(self):
+        # TODO(aryan): Stop fighting me and just work!
```

Comment:
what is this? are you ok? :sweat_smile:

Reply:
I fought a couple of tests but some continue to fight back 😕. I'm not sure if I should go ahead poking around in our state dict converters to fix it (I think it should be okay though). When we use `load_lora_weights()` for the T5 text encoder, it fails with the following:

```
    def convert_state_dict_to_peft(state_dict, original_type=None, **kwargs):
        r"""
        Converts a state dict to the PEFT format The state dict can be from previous diffusers format (`OLD_DIFFUSERS`), or
        new diffusers format (`DIFFUSERS`). The method only supports the conversion from diffusers old/new to PEFT for now.
    
        Args:
            state_dict (`dict[str, torch.Tensor]`):
                The state dict to convert.
            original_type (`StateDictType`, *optional*):
                The original type of the state dict, if not provided, the method will try to infer it automatically.
        """
        if original_type is None:
            # Old diffusers to PEFT
            if any("to_out_lora" in k for k in state_dict.keys()):
                original_type = StateDictType.DIFFUSERS_OLD
            elif any("lora_linear_layer" in k for k in state_dict.keys()):
                original_type = StateDictType.DIFFUSERS
            else:
>               raise ValueError("Could not automatically infer state dict type")
E               ValueError: Could not automatically infer state dict type
```

WDYT I should do? cc @sayakpaul

Reply:
just don't tune T5. it never works properly. we've been trying with Flux, and it just makes crap blurry